draft-ietf-quic-multipath.md

```diff
@@ -372,11 +372,11 @@ Both endpoints, namely the client and the server, can initiate path closure,
 by sending a PATH_ABANDON frame (see {{path-abandon-frame}}) which
 requests the peer to stop sending packets with the corresponding Destination Connection ID.
 
-The sender and receiver of a PATH_ABANDON frame should not release its resources
-immediately, but SHOULD wait for at least three times the current
-Probe Timeout (PTO) interval as defined in {{Section 6.2. of QUIC-RECOVERY}}
-after the last sent packet before sending the RETIRE_CONNECTION_ID frame
-for the corresponding CID.
+When sending or receiving a PATH_ABANDON frame, endpoints SHOULD wait for at
+least three times the current Probe Timeout (PTO) interval as defined in
+{{Section 6.2 of QUIC-RECOVERY}} after the last packet was sent on the path,
+before sending the RETIRE_CONNECTION_ID frame for the corresponding Connection
+ID or ceasing to recognize late acknowledgements.
```

Comment:
This is implicit in the act of sending RETIRE_CONNECTION_ID. As you has you have send that frame you can remove the CID. If you don't have the CID anymore you can't recognised any multipath-specific frame anymore because all three frames (PATH_STATUS, ACK_MP, and PATh_ABONDON) contain the CID. Instead of specifically talking about ceasing to recognise acks, maybe we can instead add a separate sentence to make that dependency more explicit.

Reply:
I think your argument is against the original text rather than the what I tried to clarify here.

The original text says that resources can be discarded at PATH_ABANDON + 3 PTO, without specifying what that resource exactly is.

My interpretation is that loss recovery is the only thing that an endpoint could possibly discard, as it has to continue decrypting packets and sending acks until it receives RETIRE_CONNECTION_ID.

Hence the proposal to be clear on that fact.

That said, I think this advice on releasing some resources, whatever they may be, is merely an optimization. I would be equally fine with removing "or ceasing to recognize late acknowledgements" entirely, as loss recovery logic can be discarded anytime when acks aren't expected, it is not specific to this particular case.